prov/efa/src/rdm/efa_rdm_pke_utils.c

```diff
@@ -275,21 +313,13 @@ int efa_rdm_pke_copy_payload_to_cuda(struct efa_rdm_pke *pke,
 	ep = pke->ep;
 	assert(ep);
 
-	ret = efa_rdm_ep_use_p2p(ep, desc);
-	if (ret < 0)
-		return ret;
-
-	segment_offset = efa_rdm_pke_get_segment_offset(pke),
-
-	p2p_available = ret;
-	local_read_available = p2p_available && efa_rdm_ep_support_rdma_read(ep);
-	cuda_memcpy_available = ep->cuda_api_permitted;
-	gdrcopy_available = desc->peer.flags & OFI_HMEM_DATA_DEV_REG_HANDLE;
+	segment_offset = efa_rdm_pke_get_segment_offset(pke);
 
-	/* For in-order aligned send/recv, only allow local read to be used to copy data */
-	if (ep->sendrecv_in_order_aligned_128_bytes) {
-		cuda_memcpy_available = false;
-		gdrcopy_available = false;
+	ret = efa_rdm_pke_get_available_copy_methods(
+		ep, desc, &local_read_available, &gdrcopy_available, &cuda_memcpy_available);
+	if (ret < 0) {
+		EFA_WARN(FI_LOG_CQ, "p2p is not supported by device\n");
```

Comment:
This error message isn't correct, p2p_support return < 0 doesn't mean p2p is not supported. It means that there is unexpected error happens during p2p check. For here, just log as `Failed to get available copy method, ret %d\n`. Change FI_LOG_CQ to FI_LOG_EP_DATA